### PR TITLE
Avoid using global Github client

### PR DESCRIPTION
**Pull request by hmarr**
Until now the global Github client has been used all over the codebase, and was pulling the access token from an environment variable with Prius. Obviously this isn't desirable if you're going to use bump as a library.

This removes the global Github client, and passes a client in from the workers, which are part of the app, not the library.

To make this change more sane, the code in `Dependency` that was hitting Github to find the changelog URL, etc., has been moved into the `DependencySourceCodeFinders` classes, which have been renamed to `DependencyMetadataFinders`.

### CODE REVIEW

**Comment by hmarr** [comment]
@hmac this should be the last meaty refactor! Feel free to ping any questions over.

In due course there's plenty more to be cleaned up in this code, but I didn't want to do it all at once.

File: lib/bump/pull_request_creator.rb [hunk 1]
```diff
@@ -80,35 +81,46 @@ def pr_name
...
       "bump_#{dependency.name}_to_#{dependency.version}"
     end
+
+    def dependency_metadata_finder
+      @dependency_metadata_finder ||=
+        begin
+          finder_class =
+            DependencyMetadataFinders.for_language(dependency.language)
+          finder_class.new(dependency: dependency, github_client: github_client)
+        end
```
Comment (hmac) [inline lib/bump/pull_request_creator.rb]: Any particular reason you've used `begin...end` here vs just chaining the methods?

Comment (greysteil) [inline lib/bump/pull_request_creator.rb]: It's a hangover from copying this method across from `Dependency`, before we moved to having a `for_language` method on the module. I'll update now - would be awesome to get this merged :)

**Comment by greysteil** [comment]
🎉 Thanks @hmac

**Pull request merged**